var/spack/repos/builtin/packages/gslib/package.py

```diff
@@ -0,0 +1,88 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Gslib(Package):
+    """Highly scalable Gather-scatter code with AMG and XXT solvers"""
+
+    homepage = "https://github.com/gslib/gslib"
+    url      = "https://github.com/gslib/gslib"
+
+    version('1.0.1', git='https://github.com/gslib/gslib.git',
+        commit='d16685f24551b7efd69e58d96dc76aec75239ea3')
+    version('1.0.0', git='https://github.com/gslib/gslib.git',
+        commit='9533e652320a3b26a72c36487ae265b02072cd48')
+
+    variant('mpi', default=True, description='Build with MPI')
+    variant('mpiio', default=True, description='Build with MPI I/O')
+    variant('blas', default=False, description='Build with BLAS')
+
+    depends_on('mpi', when="+mpi")
+    depends_on('mpi', when="+mpiio")
+    depends_on('blas', when="+blas")
+
+    def install(self, spec, prefix):
+        srcDir = 'src'
+        libDir = 'lib'
+        libname = 'libgs.a'
+
+        if self.version == Version('1.0.1'):
+            makeFile = 'Makefile'
+        else:
+            makeFile = 'src/Makefile'
+
+        CC  = self.compiler.cc
+
+        if '+mpiio' not in spec:
+            filter_file(r'MPIIO.*?=.*1', 'MPIIO = 0', makeFile)
+
+        if '+mpi' in spec:
```

Comment:
I think this should be `if '+mpi' in spec or '+mpiio' in spec` - or at least it appears based on line 65 that `+mpiio` requires `+mpi`

Reply:
You are right I will fix that.

Reply:
On a second thought, since `mpiio` depends on `mpi`, I think it is okay as it is? (because you can't have
`mpiio=True` and `mpi=False`)

Reply:
@thilinarmtb I don't think that constraint is currently expressed here: If you add `conflicts('~mpi', when='+mpiio')` then Spack will report an error when the user tries something like `spack spec gslib~mpi+mpiio`. Otherwise it would be okay with that.

Reply:
Thanks for the feedback. I will fix that. I think that is the correct way to do it, adding a conflict.

Reply:
I was under the impression that `depends_on('mpi', when="+mpiio")` will take care of this.
Thanks again !!

Reply:
I added an conflicts statement.